src/components/CMessageRow.vue

```diff
@@ -82,21 +89,24 @@ export default {
       if (!this.startedDatetime || !this.enqueuedDatetime) {
         return null;
       }
-      return formatDistance(this.enqueuedDatetime, this.startedDatetime);
+      const diff = utils.timeUnitsBetween(this.enqueuedDatetime, this.startedDatetime);
+      return `${diff.hours}:${diff.minutes}:${diff.seconds}`;
+    },
+    remainingTime() {
+      if (!this.progress) {
+        return null;
+      }
+      const factor = (1 - this.progress) / this.progress;
+      const diff = utils.timeUnitsBetween(this.startedDatetime * factor, new Date() * factor);
```

Comment:
are you sure about this ? 
you should calculate diff between Now and computed end datetime `utils.timeUnitsBetween(new date(), this.startedDatetime * factor)`